lib/vernier/output/firefox.rb

```diff
@@ -385,6 +385,20 @@ def string_table
 
         private
 
+        def pretty_name(name)
+          if name.empty?
+            tr = ObjectSpace._id2ref(@ruby_thread_id)
+            name = tr.inspect if tr
+          end
+          return name unless name.start_with?("#<Thread")
+          pretty = []
+          obj_address = name[/Thread:(0x\d+)/,1]
+          best_id = name[/\#<Thread:0x\w+@?\s?(.*)\s+\S+>/,1]
```

Comment:
i originally wanted to do this just splitting on whitespace, but realized that this could get complicated very fast:

- the thread name itself can contain spaces
- the file path can contain spaces
- both of these are optional

So instead, I try to just anchor on what we **know** will be present at the start (`#<Thread:ADDRESS`), and end (` STATE>`)

If the thread was named, then the match for `@?` should catch this. Otherwise, try to just capture everything between the start and end anchors.